src/account/AccountFactory.sol

```diff
@@ -0,0 +1,85 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.19;
+
+import {IEntryPoint} from "@eth-infinitism/account-abstraction/interfaces/IEntryPoint.sol";
+import {ERC1967Proxy} from "@openzeppelin/contracts/proxy/ERC1967/ERC1967Proxy.sol";
+import {Create2} from "@openzeppelin/contracts/utils/Create2.sol";
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+
+import {UpgradeableModularAccount} from "../../src/account/UpgradeableModularAccount.sol";
+import {ValidationConfigLib} from "../../src/helpers/ValidationConfigLib.sol";
+import {SingleSignerValidation} from "../../src/plugins/validation/SingleSignerValidation.sol";
+
+contract AccountFactory is Ownable {
+    UpgradeableModularAccount public accountImplementation;
+    bytes32 private immutable _PROXY_BYTECODE_HASH;
+    uint32 public constant UNSTAKE_DELAY = 1 weeks;
+    IEntryPoint public entryPoint;
+    address public self;
+
+    constructor(IEntryPoint _entryPoint, UpgradeableModularAccount _accountImpl) {
+        entryPoint = _entryPoint;
+        _PROXY_BYTECODE_HASH = keccak256(
+            abi.encodePacked(type(ERC1967Proxy).creationCode, abi.encode(address(accountImplementation), ""))
+        );
+        accountImplementation = _accountImpl;
+        self = address(this);
+    }
+
+    /**
+     * Create an account, and return its address.
+     * Returns the address even if the account is already deployed.
+     * Note that during user operation execution, this method is called only if the account is not deployed.
+     * This method returns an existing account address so that entryPoint.getSenderAddress() would work even after
+     * account creation
+     */
+    function createAccount(
+        address owner,
+        uint256 salt,
+        uint32 entityId,
```

Comment:
We could opt to always use an `entityId` of 0 on the factory by default, to make it simpler on the SDK

Reply:
Hmm this shouldn't be in the SDK right? I think we could always have the SDK default to 0 too, but allow flexibility in the contracts

Reply:
What would the flexibility be needed for, though? Even if it were necessary to do something else, that could just be set after deployment.